src/pages/WorkloadDetails/WorkloadInfo.tsx

```diff
@@ -223,14 +270,55 @@ class WorkloadInfo extends React.Component<WorkloadInfoProps, WorkloadInfoState>
       <>
         Pods ({pods.length}){' '}
         {validationChecks.hasPodsChecks
-          ? getValidationIcon(
+          ? getWorkloadValidationIcon(
               pods.map(a => a.name),
               'pod'
             )
           : undefined}
       </>
     );
 
+    const istioConfigItems = this.state.workloadIstioConfig ? toIstioItems(this.state.workloadIstioConfig) : [];
+    let istioTabTitle: JSX.Element | undefined;
+    let istioConfigIcon = undefined;
+    if (this.state.workloadIstioConfig?.validations) {
+      const typeNames: { [key: string]: string[] } = {};
+      if (this.state.workloadIstioConfig.validations['gateway']) {
+        this.state.workloadIstioConfig.gateways?.forEach(gw => typeNames['gateway'].push(gw.metadata.name));
```

Comment:
In the other if statements, you set the `typeNames` to an empty array. The lack of it for the gateways is intended?

Reply:
Yes, it's filtering in the UI side the validations that belongs to the Istio Config resources that are linked with a Workload.
So an empty  names array is part of the logic.